packages/manager/src/managers/BaseManager.ts

```diff
@@ -0,0 +1,92 @@
+import { SliceMachinePluginRunner } from "@slicemachine/plugin-kit";
+import { PrismicAuthManager } from "../auth/PrismicAuthManager";
+
+import { SliceMachineManager } from "./SliceMachineManager";
+
+import { UserManager } from "./user/UserManager";
+import { PrismicRepositoryManager } from "./prismicRepository/PrismicRepositoryManager";
+
+import { PluginsManager } from "./plugins/PluginsManager";
+
+import { ProjectManager } from "./project/ProjectManager";
+import { CustomTypesManager } from "./customTypes/CustomTypesManager";
+import { SlicesManager } from "./slices/SlicesManager";
+import { SnippetsManager } from "./snippets/SnippetsManager";
+import { ScreenshotsManager } from "./screenshots/ScreenshotsManager";
+import { SimulatorManager } from "./simulator/SimulatorManager";
+
+import { VersionsManager } from "./versions/VersionsManager";
+import { TelemetryManager } from "./telemetry/TelemetryManager";
+
+export abstract class BaseManager {
+	private _sliceMachineManager: SliceMachineManager;
+
+	constructor(sliceMachineManager: SliceMachineManager) {
+		this._sliceMachineManager = sliceMachineManager;
+	}
+
+	protected get prismicAuthManager(): PrismicAuthManager {
+		return this._sliceMachineManager.getPrismicAuthManager();
+	}
+
+	protected get sliceMachinePluginRunner():
+		| SliceMachinePluginRunner
+		| undefined {
+		return this._sliceMachineManager.getSliceMachinePluginRunner();
+	}
+
+	protected set sliceMachinePluginRunner(
+		sliceMachinePluginRunner: SliceMachinePluginRunner | undefined,
+	) {
+		// @ts-expect-error - _sliceMachinePluginRunner is private. We
+		// are intentially ignoring its privacy to allow Manager
+		// classes to access a shared plugin runner via protected
+		// getters and setters.
+		this._sliceMachineManager._sliceMachinePluginRunner =
+			sliceMachinePluginRunner;
+	}
+
+	protected get cwd(): string {
+		return this._sliceMachineManager.cwd;
+	}
+
+	// Protected instance prevents circular intellisense
```

Comment:
Indeed, it can! This is what autocomplete looks like if you remove the `protected` keyword:

<img width="641" alt="image" src="https://user-images.githubusercontent.com/8601064/220801953-7b57ff43-a444-462a-a774-747493e2c0fe.png">

`user` is a valid option, so it lets to access it. If we make it `protected`, you can only access it one level deep.